Sources/Networking/Operations/GetProductsEntitlementsOperation.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  GetProductsEntitlementsOperation.swift
+//
+//  Created by Nacho Soto on 3/17/23.
+
+import Foundation
+
+final class GetProductsEntitlementsOperation: CacheableNetworkOperation {
+
+    private let callbackCache: CallbackCache<ProductsEntitlementsCallback>
+
+    static func createFactory(
+        configuration: NetworkConfiguration,
+        callbackCache: CallbackCache<ProductsEntitlementsCallback>
+    ) -> CacheableNetworkOperationFactory<GetProductsEntitlementsOperation> {
+        return .init({ cacheKey in
+                .init(
+                    configuration: configuration,
+                    callbackCache: callbackCache,
+                    cacheKey: cacheKey
+                )
+        },
+                     individualizedCacheKeyPart: "")
+    }
+
+    private init(configuration: NetworkConfiguration,
+                 callbackCache: CallbackCache<ProductsEntitlementsCallback>,
+                 cacheKey: String) {
+        self.callbackCache = callbackCache
+        super.init(configuration: configuration, cacheKey: cacheKey)
+    }
+
+    override func begin(completion: @escaping () -> Void) {
+        self.getResponse(completion: completion)
+    }
+
+}
+
+private extension GetProductsEntitlementsOperation {
+
+    func getResponse(completion: @escaping () -> Void) {
+        let request = HTTPRequest(method: .get, path: .getProductsEntitlements)
```

Comment:
I'm not sure but we might have a different base URL for this one @aboedo? Or would we just use DNS to point to whatever server we want to? If we need a different base URL, we will need to add support for that (in Android, that was added for diagnostics). In any case, that should be done in a different PR.

Reply:
Looks like the goal is to have it as the same domain, but it's not built yet.